### PR TITLE
Fix for redbox on read message

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 

The old code had a section for handling non userInteraction (yet didnt' actually check that flag) then had a section for if userInteraction was true. Turns out its possible to get readmessage calls w/ userInteraction flag on, which this didn't expect so that flow would

1. Handle it in the first part of the code
2. Fail to return
3. Go into the if (payload.userInteraction) code and then have a type it doesn't expect and red box

Instead of adding the extra return I changed this to be a switch statement with more explicit bailouts

cc: @mmaxim

### CODE REVIEW

File: shared/actions/push.js [hunk 1]
```diff
@@ -79,11 +79,25 @@ const resetHandledPush = () => {
 function* pushNotificationSaga(notification: PushGen.NotificationPayload): Saga.SagaGenerator<any, any> {
   logger.info('Push notification:', notification)
   const payload = notification.payload.notification
-  if (payload) {
-    // Handle types that are not from user interaction
```
Comment (chrisnojima) [inline shared/actions/push.js]: note it doesn't actually check

File: shared/actions/push.js [hunk 2]
```diff
@@ -98,40 +112,30 @@ function* pushNotificationSaga(notification: PushGen.NotificationPayload): Saga.
...
       } catch (err) {
-        logger.info('failed to unbox silent notification', err)
+        logger.error('failed to unbox silent notification', err)
       }
-      return
-    } else if (payload.type === 'chat.readmessage') {
-      logger.info('Push notification: read message notification received')
-      const b = typeof payload.b === 'string' ? parseInt(payload.b) : payload.b
-      if (b === 0) {
-        clearAllNotifications()
```
Comment (chrisnojima) [inline shared/actions/push.js]: note the lack of return here which caused the problem in the old code

**Comment by chrisnojima** [comment]
this diff is kinda hard to read so maybe better to look at the code side by side

File: shared/actions/push.js [hunk 3]
```diff
@@ -143,18 +147,29 @@ function* pushNotificationSaga(notification: PushGen.NotificationPayload): Saga.
...
         yield Saga.put(createShowUserProfile({username}))
-      } else {
-        logger.error('Push notification payload missing or unknown type')
+      } catch (err) {
+        logger.error('failed to handle follow push', err)
       }
-    }
+      break
+    default:
+      logger.error('Push notification payload missing or unknown type')
```
Comment (akalin-keybase) [inline shared/actions/push.js]: Maybe log `JSON.stringify(payload)`, too

**Comment by chrisnojima** [comment]
i'm merging this monday morning

**Pull request merged**